contracts/strategies/allocation/AllocationWithVotingBase.sol

```diff
@@ -226,28 +235,41 @@ contract AllocationWithVotingBase is IAllocationStrategy, Ownable, Initializable
         }
 
         // Decode the _data
-        Allocations[] memory allocationsPerToken = abi.decode(_data, Allocations[]);
+        (Allocations[] memory allocationsPerToken) = abi.decode(_data, (Allocations[]));
         for (uint256 i; i < allocationsPerToken.length;) {
             // Get the allocations per token
             Allocations memory allocations = allocationsPerToken[i];
 
             uint256 allocationsTotalAmount = allocations.totalAmount;
 
+            // convert to TransferData
+            Transfer.TransferData memory transferData =
```

Comment:
not clear why we are converting this here.  Do we need this ?

Reply:
Yes, the `Transfer.tranferAmounts()` takes a `TransferData` struct.

`function transferAmount(address _token, TransferData memory _transferData)`
`function transferAmounts(address _token, TransferData[] memory _transferData)`

```javacript
struct Allocations {
    address token;
    uint256 totalAmount;
    Allocation[] data;
}

struct Allocation {
    uint256 applicationId;
    uint256 amount;
}

struct TransferData {
    address to;
    uint256 amount;
}
```

We either send an array of `TransferData` or a single `TransferData` object.

Reply:
Ah makes sense

Reply:
Gotcha